libraries/redcore/api/soap/operation/operation.php

```diff
@@ -61,63 +55,60 @@ public function readList(
 			$dataGet = JArrayHelper::fromObject($dataGet);
 		}
 
-		if ($limitStart != 0)
-		{
-			$dataGet['list']['limitstart'] = (int) $limitStart;
-		}
-
-		if ($limit != 20)
-		{
-			$dataGet['list']['limit'] = (int) $limit;
-		}
+		$dataGet['list']['limitstart'] = (isset($data->limitStart) ? (int) $data->limitStart : 0);
+		$dataGet['list']['limit'] = (isset($data->limit) ? (int) $data->limit : 20);
+		$dataGet['filter']['search'] = (isset($data->filterSearch) ? (string) $data->filterSearch : '');
 
-		if (!is_null($filterSearch))
-		{
-			$dataGet['filter']['search'] = $filterSearch;
-		}
-
-		if (!empty($filters) && is_array($filters))
-		{
-			foreach ($filters as $key => $value)
-			{
-				$dataGet['filter'][$key] = $value;
-			}
-		}
+		$filters = RApiHalHelper::getFilterFields($this->webservice->configuration->operations->read->list);
 
-		if (!is_null($ordering))
+		foreach ($filters as $filter)
 		{
-			$dataGet['filter']['order'] = $ordering;
+			$dataGet['filter'][$filter] = $data->filters->$filter;
```

Comment:
yes and getFilterFields only returns the filter names when $fullDefinition = false (the default for that function)
that's why it works :)